src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java

```diff
@@ -187,6 +190,19 @@ public List<Dataset> createDatasetsFromRegistration(
             datasetInserts);
 
     List<Dataset> datasets = datasetDAO.findDatasetsByIdList(createdDatasetIds);
+    try {
```

Comment:
Not sure if this is the best approach

Reply:
There are a couple ways to test. If your local email config is set to not send email (which should be the case), then no email will be sent BUT the content should be saved to the email table. There is also the `test-sendgrid-html.sh` script that might be helpful to see the final output.

Reply:
What I am not sure about:

1. Whether this is the best approach to notify DAC chairs that a new dataset has been submitted
2. How to best add unit tests for this specific code chunk

All the other pieces are tested (including sending the template with the appropriate details)

Reply:
To answer specifically:

1. Looking more closely at this, I think it needs to go down one level into `DatasetServiceDAO.insertDatasetRegistration`. I say that because there are two code paths for dataset creation - one of them in this code path where we are creating a brand new dataset registration. The other happens during the Study update process where the user could be adding new consent groups (datasets) to a study. Both processes use `DatasetServiceDAO.insertDatasetRegistration`. The alternative is to add this block to both the create and update code blocks which might be a good idea if ... 
2. I would suggest pulling this out into a separate method so it can be tested independently of dataset creation. If this block were outside of this larger method, you can easily mock the dao and service calls and ensure that `sendDatasetSubmittedMessage` is called an appropriate number of times given the dataset and dac chairs. If it is a separate method, it can be called conveniently from both the create and the update code paths.